rdmo/accounts/models.py

```diff
@@ -123,6 +130,64 @@ class Meta:
     def __str__(self):
         return self.user.username
 
+    @classmethod
+    def create_consent(cls, user, session=None) -> bool:
+        obj, _created = cls.objects.update_or_create(user=user, defaults={"consent": True})
+
+        # Validate consent before storing in session
+        if obj.is_consent_valid():
+            if session:
+                session[CONSENT_SESSION_KEY] = True
+            return True
+
+        obj.delete()  # Remove when consent is outdated
+        return False
+
+
+    @classmethod
+    def has_accepted_terms(cls, user, session) -> bool:
+        if not settings.ACCOUNT_TERMS_OF_USE:
+            return True  # If terms are disabled, assume accepted.
+
+        # Check session cache first
+        if CONSENT_SESSION_KEY in session:
+            return session[CONSENT_SESSION_KEY]
+
+        # Query the database if not cached
+        consent = cls.objects.filter(user=user).first()
+        has_valid_consent = bool(consent and consent.is_consent_valid())
+
+        session[CONSENT_SESSION_KEY] = has_valid_consent  # Cache result
+        return has_valid_consent
+
+    def is_consent_valid(self) -> bool:
+        # optionally enable terms to be outdated
+        terms_version_date = getattr(settings, 'TERMS_VERSION_DATE', None)
```

Comment:
As a flow for invalidating the consent (eg. when the text needs an update or the terms changed), 
I thought it would be helpful to be able to simply set this date as a setting instead of manually deleting all of the `ConsentFieldValue` objects. 
Is the name of the `TERMS_VERSION_DATE` ok? Should it be added to the normal core/settings as `TERMS_VERSION_DATE = None` ?

Reply:
I like it! And (see my comment above) should have a sensible default (`None`) in `rdmo.core.settings`.

Reply:
Line 166 can be removed.

Reply:
Actually I would call it `ACCOUNT_TERMS_OF_USE_DATE` to have some kind of `ACCOUNT_TERMS_OF_USE_` namespacing in the settings.

Reply:
Same for the other terms of use related settings.

Reply:
I found now a way to validate a setting via the `from django.core.checks import Error, register`. Do you mind if I add an `rdmo.core.apps.py`  (for the `ready` method) and a `rdmo.core.checks.py` that validates this date string setting?
Im not sure if it's too much or if it could be useful to implement checks also for other settings in this way.

Reply:
I would stick to having a default value for the settings used in RDMO (otherwise we are inconsistent in `rdmo`), but something like this could be nice for plugins.

Reply:
yes, I mean in addition to setting the default. The default is `None` and then when another value is set, it needs to be a valid to-a-date-parse-able string.

Reply:
Hmm, I would not start checking `settings`. If RDMO is misconfigured exceptions will happen. If you mean to check validate the settings *once* when starting and outside of the usage of the settings in the code, I am fine with it. But I don't want to have `validate(settings.FOO_BAR)` everywhere.

Reply:
yes, it's exactly that, for `manage.py check`:

```
SystemCheckError: System check identified some issues:

ERRORS:
?: (core.E001) ACCOUNT_TERMS_OF_USE_DATE = 2025-02-36 is not a valid date string.
        HINT: day is out of range for month

System check identified 1 issue (0 silenced).
❯
```

Reply:
Awesome, go for it.